app.rb

```diff
@@ -0,0 +1,131 @@
+
+class App
+  def initialize
+    @books = []
+    @persons = []
+    @rentals = []
+  end
+
+  def start_console
+    puts 'Welcome to my School Library!'
+    until list_of_options
+      input = gets.chomp
+      if input == '7'
+        puts 'Thank you for using my School Library!'
+        break
+      end
+
+      option input
+    end
+  end
+
+  def list_all_books
+    puts 'Database is empty! Add a book.' if @books.empty?
+    @books.each { |book| puts "[Book] Title: #{book.title}, Author: #{book.author}" }
+  end
+
+  def list_all_persons
+    puts 'Database is empty! Add a person.' if @persons.empty?
+    @persons.each { |person| puts "[#{person.class.name}] Name: #{person.name} Age: #{person.age} Id: #{person.id}" }
+  end
+
+  def create_person
+    print 'To create a student, press 1, to create a teacher, press 2 :'
+    option = gets.chomp
+
+    case option
+    when '1'
+      create_student
+    when '2'
+      create_teacher
+    else
+      puts 'Invalid input. Try again'
+    end
+  end
+
+  def create_student
+    puts 'Create a new student'
+    print 'Enter name: '
+    name = gets.chomp
+    print 'Enter student age: '
+    age = gets.chomp.to_i
+    print 'Has parent permission? [Y/N]: '
+    parent_permission = gets.chomp.downcase
+    case parent_permission
+    when 'n'
+      Student.new(age, 'classroom', name, parent_permission: false)
+      @persons << student
+      puts "Student doesn't have parent permission, can't rent a books "
+    when 'y'
+      student = Student.new(age, 'classroom', name, parent_permission: false)
+      @persons << student
+      puts 'Student created successfully'
+    end
+  end
+
+  def create_teacher
+    puts 'Create a new teacher'
+    print 'Enter name: '
+    name = gets.chomp
+    print 'Enter teacher age'
+    age = gets.chomp.to_i
+    print 'Enter teacher specialization: '
+    specialization = gets.chomp
+    teacher = Teacher.new(age, specialization, name)
+    @person << teacher
+    puts 'Teacher created successfully'
+  end
+
+  def create_book
+    puts 'Create a new book'
+    print 'Enter title: '
+    title = gets.chomp
+    Print 'Enter an author: '
+    author = gets.chomp
+    book = Book.new(title, author)
+    @books.push(book)
+    puts "Book #{title} created successfully."
+  end
+
+  def create_rental
+    puts 'Select which book you want to rent by entering its number'
+    @books.each_with_index { |_book, index| puts "[#{index}] Title: #{books.title} Author: #{books.author}" }
+
+    book_id = gets.chomp.to_i
+
+    Puts 'Select a person from the list by its number'
+    @person.each_with_index do |person, index|
+      puts "[#{index}] [#{person.class.name}] Name: #{person.name} Id: #{person.id} Age: #{person.age}"
+    end
+
+    person_id = gets.chomp.to_i
+
+    print 'Date: '
+    date = gets.chomp.to_s
+
+    rental = Rental.new(date, @persons[person_id], @books[book_id])
+    @rentals << rental
+
+    puts 'Rental created successfully'
+  end
```

Comment:
I am unable to create a new rental because your `create_rental` method throws for a few reasons

- [ ] you have a typo where you have written `Puts` which is not a valid method instead of `puts`.
- [ ] you are iterating through the `@books` however you are using the variable `books.title` and `books.author`. this can't work because `books` is not defined. what you should use is the parameter `_book` (note: you should also rename `_book` to simply `book` because there is no real need to prefix it with an underscore.
- [ ] you are trying to iterate through `@persons` however you have a typo where you wrote `@person`.

Reply:
Thank you for the feedback. I fixed the bugs.